framework/include/timeintegrators/TimeIntegrator.h

```diff
@@ -127,6 +127,11 @@ class TimeIntegrator : public MooseObject,
    */
   virtual bool isExplicit() const { return false; }
 
+  /*
+   * Returns whether the time integrator controls its own state
```

Comment:
I would make this doxygen more clear and verbose. I might even rename the method to `advancesProblemState()`